funsor/ops.py

```diff
@@ -347,3 +347,14 @@ def reciprocal(x):
     'truediv',
     'xor',
 ]
+
+__doc__ = """
+Built-in operations
+-------------------
+
+{}
+
+Operation classes
+-----------------
+""".format("\n".join(f".. autodata:: {_name}\n"
+                     for _name in __all__ if isinstance(globals()[_name], Op)))
```

Comment:
Neat - should do something similar for patterns/interpretations?  We could add docstrings to (important) patterns and auto-generate separate doc pages for each interpretation.

Reply:
Hmm, interesting. I'll have to think about how to do that. @jpchen had suggested making automatically making latex tables as in our paper.